homeassistant/auth/__init__.py

```diff
@@ -83,6 +116,15 @@ def auth_providers(self):
         """Retrieve a user."""
         return await self._store.async_get_user(user_id)
 
+    async def async_get_user_by_credentials(self, credentials):
+        """Get a user by credential, raise ValueError if not found."""
```

Comment:
Doc string is not correct.

Reply:
Generaly, if we're querying a value, `None` is an acceptable return value. If we're mutating, `ValueError` if the input value is incorrect.